graphql/schemabuilder/input.go

```diff
@@ -316,7 +316,7 @@ func (sb *schemaBuilder) makeSliceParser(typ reflect.Type) (*argParser, graphql.
 	return &argParser{
 		FromJSON: func(value interface{}, dest reflect.Value) error {
 			asSlice, ok := value.([]interface{})
-			if !ok {
+			if asSlice != nil && !ok {
```

Comment:
I would have thought that if the type conversion fails, we wouldn't care about the value of `asSlice`.  Is this really necessary?